src/main/java/taxi/dao/impl/ManufacturerDaoJdbcImpl.java

```diff
@@ -0,0 +1,120 @@
+package taxi.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import taxi.dao.ManufacturerDao;
+import taxi.lib.Dao;
+import taxi.model.Manufacturer;
+import taxi.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoJdbcImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String sqlQuery = "INSERT INTO manufacturer (manufacturer_name, manufacturer_country)"
+                + "VALUES (?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(sqlQuery, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, manufacturer.getName());
+            preparedStatement.setString(2, manufacturer.getCountry());
+            preparedStatement.executeUpdate();
+            ResultSet resultset = preparedStatement.getGeneratedKeys();
+            while (resultset.next()) {
+                manufacturer.setId(resultset.getObject(1, Long.class));
+            }
+            preparedStatement.close();
+            connection.close();
```

Comment:
If you use "try with resources", you don't need to close them after

Reply:
Really, I think same,but  in checklist "Best practices with closing Connections and/or PreparedStatements...". Ok, i agree.